### PR TITLE
Restructuring logs_dir to be parent folder of "subagents"

### PR DESCRIPTION
**Pull request by rafaelwestphal**
## Description
Follow up to PR #1253 where the restructuring was suggested
Now that we have health-checks.log in addition to logging-module.log, the logs_dir should point to the parent of "subagents" in linux. The location of the files will be the same, but it makes referencing the directories easier.

## Related issue
b/283253077

## Checklist:
- Unit tests
  - [ ] Unit tests do not apply.
  - [X] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [X] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [X] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [X] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: systemd/google-cloud-ops-agent-fluent-bit.service.d/directories.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # For distros with systemd prior to version 240:
 [Service]
-Environment=RUNTIME_DIRECTORY=%t/google-cloud-ops-agent-fluent-bit STATE_DIRECTORY=/var/lib/google-cloud-ops-agent/fluent-bit LOGS_DIRECTORY=/var/log/google-cloud-ops-agent/subagents
-ExecStartPre=/bin/mkdir -p ${RUNTIME_DIRECTORY} ${STATE_DIRECTORY} ${LOGS_DIRECTORY}
+Environment=RUNTIME_DIRECTORY=%t/google-cloud-ops-agent-fluent-bit STATE_DIRECTORY=/var/lib/google-cloud-ops-agent/fluent-bit LOGS_DIRECTORY=/var/log/google-cloud-ops-agent
+ExecStartPre=/bin/mkdir -p ${RUNTIME_DIRECTORY} ${STATE_DIRECTORY} ${LOGS_DIRECTORY}/subagents
```
Comment (franciscovalentecastro) [inline systemd/google-cloud-ops-agent-fluent-bit.service.d/directories.conf:4]: Should this also create `${LOGS_DIRECTORY}` by itself before creating the `..../subagents` subfolder ?

Comment (rafaelwestphal) [reply systemd/google-cloud-ops-agent-fluent-bit.service.d/directories.conf:4]: The -p flag creates the parent aswell.

File: confgenerator/confgenerator.go [hunk 2]
```diff
@@ -381,14 +382,14 @@ func (l *Logging) generateFluentbitComponents(ctx context.Context, userAgent str
...
-		IncludePaths: []string{"${logs_dir}/logging-module.log"},
+		IncludePaths: []string{fluentbitSelfLogs(platform.FromContext(ctx))},
 		//Following: b/226668416 temporarily set storage.type to "memory"
 		//to prevent chunk corruption errors
 		BufferInMemory: true,
 	}.Components(ctx, fluentBitSelfLogTag)...)
 
 	out = append(out, LoggingReceiverFilesMixin{
-		IncludePaths:   []string{healthChecksLog(platform.FromContext(ctx))},
+		IncludePaths:   []string{path.Join("${logs_dir}", "health-checks.log")},
```
Comment (franciscovalentecastro) [inline confgenerator/confgenerator.go:392]: Maybe we can also define a constant for `health-checks.log` or a function like the one for fluent-bit.

File: confgenerator/confgenerator.go [hunk 3]
```diff
@@ -244,11 +244,12 @@ func contains(s []string, str string) bool {
 	return false
 }
 
-func healthChecksLog(p platform.Platform) string {
+func fluentbitSelfLogs(p platform.Platform) string {
```
Comment (franciscovalentecastro) [inline confgenerator/confgenerator.go:247]: [Optional] Rename `fluentbitSelfLogsPath`.

**Review by franciscovalentecastro — APPROVED** [review]

**Pull request merged**